src/mlpack/core/data/bow_encoding.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file bow.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of Bag of Words encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_BOW_ENCODING_HPP
+#define MLPACK_CORE_DATA_BOW_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <utility>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Bow Enocding class
```

Comment:
perhaps it makes sense to include a simple example here as well, something like:


https://github.com/mlpack/mlpack/blob/897b0cddf6ba23733f701b4679fac08f9f90ebb2/src/mlpack/core/data/binarize.hpp#L26-L34